core/src/services/libsql/backend.rs

```diff
@@ -0,0 +1,345 @@
+        };
+        let root = normalize_root(
+            self.root
+                .clone()
+                .unwrap_or_else(|| "/".to_string())
+                .as_str(),
+        );
+
+        Ok(LibsqlBackend::new(Adapter {
+            connection_string: conn,
+            auth_token: self.auth_token.clone(),
+            table,
+            key_field,
+            value_field,
+        })
+        .with_root(&root))
+    }
+}
+
+/// Backend for libsql service
+pub type LibsqlBackend = kv::Backend<Adapter>;
+
+#[derive(Clone)]
+pub struct Adapter {
+    connection_string: String,
+    auth_token: Option<String>,
+
+    table: String,
+    key_field: String,
+    value_field: String,
+}
+
+impl Debug for Adapter {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut ds = f.debug_struct("LibsqlAdapter");
+        ds.field("connection_string", &self.connection_string)
+            .field("table", &self.table)
+            .field("key_field", &self.key_field)
+            .field("value_field", &self.value_field);
+
+        if self.auth_token.is_some() {
+            ds.field("auth_token", &"<redacted>");
+        }
+
+        ds.finish()
+    }
+}
+
+impl Adapter {
+    fn get_config(&self) -> Result<Config> {
+        let mut config = Config::new(self.connection_string.clone().as_str()).map_err(|err| {
+            Error::new(ErrorKind::ConfigInvalid, "connection_string is invalid")
+                .with_context("service", Scheme::Libsql)
+                .set_source(err)
+        })?;
+        if let Some(auth_token) = self.auth_token.clone() {
+            config = config.with_auth_token(auth_token);
+        }
+        Ok(config)
+    }
+}
+
+#[async_trait]
+impl kv::Adapter for Adapter {
+    fn metadata(&self) -> kv::Metadata {
+        kv::Metadata::new(
+            Scheme::Libsql,
+            &self.table,
+            Capability {
+                read: true,
+                write: true,
+                create_dir: true,
+                delete: true,
+                ..Default::default()
+            },
+        )
+    }
+
+    async fn get(&self, path: &str) -> Result<Option<Vec<u8>>> {
+        let cloned_path = path.to_string();
+        let cloned_self = self.clone();
+
+        task::spawn_blocking(move || cloned_self.blocking_get(cloned_path.as_str()))
+            .await
+            .map_err(|err| {
+                Error::new(
+                    ErrorKind::Unexpected,
+                    "unhandled error from libsql when spawning task",
+                )
+                .set_source(err)
+            })
+            .and_then(|inner_result| inner_result)
+    }
+
+    fn blocking_get(&self, path: &str) -> Result<Option<Vec<u8>>> {
+        let query = format!(
+            "SELECT {} FROM {} WHERE `{}` = ? LIMIT 1",
+            self.value_field, self.table, self.key_field
+        );
+        let client = SyncClient::from_config(self.get_config()?).map_err(|err| {
```

Comment:
Instead of creating a client for each request, store an `Arc<Mutex<Client>>` in the `Adapter`. Additionally, there is no need to manually implement `blocking_xxx`.